File: onyx-database/src/main/java/com/onyx/exception/BufferUnderflowException.java
package com.onyx.exception;

/**
 * Created by tosborn1 on 2/10/17.
 *
 */
public class BufferUnderflowException extends BufferingException {

    public static final String BUFFER_UNDERFLOW = "Buffer Underflow exception ";

    public BufferUnderflowException(@SuppressWarnings("SameParameterValue") String message) {
        super(message);
    }

    public BufferUnderflowException(@SuppressWarnings("SameParameterValue") String message, Class clazz) {
        super(message, clazz);
    }
}

File: onyx-database/src/main/kotlin/com/onyx/exception/BufferUnderflowException.kt
package com.onyx.exception

/**
 * Created by Tim Osborn on 2/10/17.
 *
 */
class BufferUnderflowException : BufferingException {

    @JvmOverloads
    constructor(message: String? = "") : super(message)

    constructor(message: String, clazz: Class<*>) : super(message, clazz)

    companion object {

        @JvmField val BUFFER_UNDERFLOW = "Buffer Underflow exception "
    }
}
